### ISSUE TITLE
Update rancher system charts to work in k8s v1.16

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#22488]
See: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/
Deprecated APIs are being removed from k8s 1.16. We need to make sure the charts in our catalogs will work with v1.16\

Relates to https://github.com/rancher/rancher/issues/22426

**Comment by deniseschannon** [comment]
The system charts to check that work on k8s 1.16 is

external-dns
logging
monitoring
istio

Just need to make sure it deploys with k8s 1.16 due to the API deprecations.

**Comment by bmdepesa** [comment]
rancher/rancher:master-head 4c7e3263a

Logging tested here: https://github.com/rancher/rancher/issues/21770

Monitoring app fails to deploy in 1.16 with error: `Failed to install app cluster-monitoring. Error: validation failed: unable to recognize “”: no matches for kind “PodSecurityPolicy” in version “extensions/v1beta1`

Istio enables successfully, and appears to function, but the Graph and Prometheus/Grafana links are unavailable as Grafana / Prometheus are not successfully deployed due to the monitoring chart needing updates.

**Comment by jiaqiluo** [comment]
See the same error as @bmdepesa's comment in `Rancher:v2.3.0-rc1`, air-gapped HA mode
 
- enable the cluster monitoring in k8s 1.16.0 cluster

<img width="1490" alt="Screen Shot 2019-09-05 at 3 22 54 PM" src="https://user-images.githubusercontent.com/6218999/64387510-105b5600-cff1-11e9-8de6-dd4bf38cb800.png">

**Comment by gitlawr** [comment]
Besides updating the API group for the newly added PSP resource, we also need to update the Prometheus operator.
see https://github.com/coreos/prometheus-operator/issues/2592

Here's the error logs from the Prometheus operator when enabling monitoring in 1.16 clusters:
```
ts=2019-09-06T03:00:39.595806765Z caller=main.go:169 msg="Starting Prometheus Operator version '0.29.0'."
level=info ts=2019-09-06T03:00:39.69272559Z caller=operator.go:213 component=alertmanageroperator msg="connection established" cluster-version=v1.16.0-beta.1
level=info ts=2019-09-06T03:00:39.693016093Z caller=operator.go:381 component=prometheusoperator msg="connection established" cluster-version=v1.16.0-beta.1
level=info ts=2019-09-06T03:00:39.693117435Z caller=operator.go:397 component=prometheusoperator msg="CRD API endpoints ready"
level=info ts=2019-09-06T03:00:39.693093665Z caller=operator.go:229 component=alertmanageroperator msg="CRD API endpoints ready"
E0906 03:00:39.701485 1 reflector.go:134] github.com/coreos/prometheus-operator/pkg/alertmanager/operator.go:237: Failed to list *v1beta2.StatefulSet: the server could not find the requested resource
E0906 03:00:39.716717 1 reflector.go:134] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:409: Failed to list *v1beta2.StatefulSet: the server could not find the requested resource
E0906 03:00:40.703712 1 reflector.go:134] github.com/coreos/prometheus-operator/pkg/alertmanager/operator.go:237: Failed to list *v1beta2.StatefulSet: the server could not find the requested resource
E0906 03:00:40.717854 1 reflector.go:134] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:409: Failed to list *v1beta2.StatefulSet: the server could not find the requested resource
```

**Comment by deniseschannon** [comment]
@gitlawr @loganhz sounds like we need to bump prometheus-operator up to the latest version. They just released v0.32.0.

**Comment by gitlawr** [comment]
I have a feeling that we need to update the embedded helm/tiller as well(which is using go-client 1.10). Haven't dug out the root cause but filed the issue for an incompatibility: https://github.com/rancher/rancher/issues/22684

**Comment by gitlawr** [comment]
Note for QA: workloads are not cleaned up on disabling before #22684 is addressed. Delete the app namespace as the workaround.

**Comment by sangeethah** [comment]
external DNS tested - https://github.com/rancher/rancher/issues/21768

**Comment by cjellick** [comment]
To @gitlawr's point, we will indeed need to upgrade our embedded helm.tiller, but that has been punted out of v2.3.0

**Comment by cjellick** [comment]
PRs merged, but we cant really fully test until we fix helm. Deciding what to do with this issue.

**Comment by deniseschannon** [comment]
@cjellick could we move this to test to be able to test out and use the other Issue about the fact that workloads are not cleaned up on disabling before #22684 would be part of k8s 1.16 testing for helm/tiller upgrade?

**Comment by cjellick** [comment]
fine by me

**Comment by jiaqiluo** [comment]
The bug fix is validated in master-head 768c4bff4

I can: 
- add an ec2 cluster with  `v1.16.0-beta.1`: 1 etcd, 1 control plane, 2 workers, k8s 
- enable the cluster monitoring 

```
> k version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.5", GitCommit:"0e9fcb426b100a2aea5ed5c25b3d8cfbb01a8acf", GitTreeState:"clean", BuildDate:"2019-08-05T09:21:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-beta.1", GitCommit:"92639f719ee353219fa2cd19128a93409bb3a1e3", GitTreeState:"clean", BuildDate:"2019-08-20T13:22:52Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by sangeethah** [comment]
@jiaqiluo Can you also test if enabling `Istio` in a new cluster (were monitoring is not enabled) works?
